src/surface_conditions/surface_conditions.jl

```diff
@@ -131,16 +136,20 @@ function surface_state_to_conditions(
 
     (!isnothing(q_vap) && atmos.moisture_model isa DryModel) &&
         error("surface q_vap cannot be specified when using a DryModel")
-
-    if isnothing(T)
+    if isnothing(sfc_prognostic_values)
```

Comment:
@dennisYatunin Will this affect the coupler behavior (I think not but want to double check)?